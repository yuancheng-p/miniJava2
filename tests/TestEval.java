public class TestEval {

  public static void main(String [] args) {
    double a = 1.2;
    double c = 1.2 + a;

    // test TNew
    TestEval te = new TestEval();

    // test TCall & TReturn
    (new TestEval()).add(te.bar(), 122);

    // test TBlock
    boolean b = true;
    {
      b = false;
      boolean v2 = true;
    }

    // test if else
    if (a > 0 && false) {
      c = a;
      int temp = 110;
    } else {
      c = a + 2;
    }

    te.test_while();
  }

  public void test_while() {
    int i = 0;
    while (i < 2) {
      i = i + 100;
    }
  }

  public int bar () {
    return 888;
  }


  public int add(int a, int b) {
    return a + b;
  }

  public void foo () {
    int f = 999;
    return;
  }
}


class Node {
  public void foobar() {
  }
}
