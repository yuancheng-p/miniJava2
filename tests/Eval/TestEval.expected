successfull parsing
start typing
check add and methods:TestEval:main
check add and methods:TestEval:test_while
check add and methods:TestEval:bar
check add and methods:TestEval:add
check add and methods:TestEval:foo
check add and methods:Node:foobar
--------- env --------
Boolean : class_env(parent=: Object)
Integer : class_env(parent=: Object)
Node : class_env(parent=: Object)
method : foobar
Object : class_env(parent=: )
String : class_env(parent=: Object)
TestEval : class_env(parent=: Object)
method : foo
method : bar
method : main paramlist: String[] 
method : add paramlist: int  paramlist: int 
method : test_while
end typing
_______________main___________________
double a = 1.2;
    /*
    ------ a in frame ------
    a : 1.2
    */

double c = 1.2+a;
    /*
    ------ c in frame ------
    c : 2.4
    */

TestEval te = new TestEval();
    /*
    ------ te in frame ------
    te : 0 (ref)
    ------ object in heap ------
    [TestEval] (0) :
    */

new TestEval().add(122,te.bar());
    /*
    ------ Call method: bar() ---------
    ------ Return: 888 ------------
    */

    /*
    ------ Call method: add() ---------
    ------ Return: 1010 ------------
    */

boolean b = true;
    /*
    ------ b in frame ------
    b : true
    */

b=false;
boolean v2 = true;
    /*
    ------ v2 in frame ------
    v2 : true
    */

c=a+2;
te.test_while();
    /*
    ------ Call method: test_while() ---------
int i = 0;
    /*
    ------ i in frame ------
    i : 0
    */

    i=i+100;
    ------ Return: void ------------
    */

